compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -1404,6 +1404,15 @@ fn compare_number_of_method_arguments<'tcx>(
                     potentially_plural_count(trait_number_args, "parameter")
                 ),
             );
+            if let Ok(snip) = tcx.sess.source_map().span_to_snippet(trait_span) {
```

Comment:
This just copies the contents of the parentheses, right? If so, this doesn't take params into account, I think:

```
trait Foo {
  fn bar<T>(_: T);
}

impl Foo for () {
  fn bar<E>(_: E, _: i32) {}
}
```

which would suggest changing `fn bar<E>(_: T) {}`

Reply:
yes it would suggest:
```
error[E0050]: method `bar` has 2 parameters but the declaration in trait `Foo::bar` has 1
 --> params.rs:6:16
  |
2 |   fn bar<T>(_: T);
  |                - trait requires 1 parameter
...
6 |   fn bar<E>(_: E, _: i32) {}
  |                ^^^^^^^^^ expected 1 parameter, found 2
  |
help: modify the signature to match the trait definition
  |
6 |   fn bar<E>(_: T) {}
```
I think we should suggest it to be `fn bar<E>(_: E) {}` or `fn bar<T>(_: T) {}`

Reply:
Similarly, we need to consider cases where the arguments reference generics from the trait:

```
trait Foo<T> {
  fn foo(_: T);
}

impl Foo<()> for i32 {
  fn foo();
}
```

Should suggest `_: ()`, not `_: T`.